src/backend/binaryen/wasm_type_gen.ts

```diff
@@ -120,13 +120,13 @@ export class WASMTypeGen {
             case ValueTypeKind.NUMBER:
             case ValueTypeKind.STRING:
             case ValueTypeKind.RAW_STRING:
-            case ValueTypeKind.NULL:
             case ValueTypeKind.UNDEFINED:
             case ValueTypeKind.UNION:
             case ValueTypeKind.ANY:
             case ValueTypeKind.INT:
                 this.createWASMBaseType(type);
                 break;
+            case ValueTypeKind.NULL:
```

Comment:
Why we treat is as empty type?

Reply:
```ts
interface I2 {
    x: A | null;
}
let i2: I2 = { x: null };
```
In `{ x: null }`, `x` propertyType should be static, and should be seen compatibled with `A | null`, so that we can do static assinment.

If  `x` propertyType is regarded as `any type`, then it will never be assigned to `A | null` type.